control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go

```diff
@@ -71,9 +75,40 @@ func ReconcileNetworkOperator(network *operatorv1.Network, networkType hyperv1.N
 
 	// Setting the management state is required in order to create
 	// this object. We need to create this object before the cno starts
-	// because mutating many of the values (like vxlanport) is not premitted
+	// because mutating many of the values (like vxlanport) is not permitted
 	// after the cno reconciles this operator CR
 	if network.Spec.ManagementState == "" {
 		network.Spec.ManagementState = "Managed"
 	}
 }
+
+func DetectSuboptimalMTU(ctx context.Context, mgmtClient client.Client,
+	guestNetworkOperator *operatorv1.Network, hcp *hyperv1.HostedControlPlane) error {
+	const recommendedMinMTU = uint32(9000)
+	const ovnkOverhead = 200
+	if hcp.Spec.Platform.Type == hyperv1.KubevirtPlatform &&
+		hcp.Spec.Networking.NetworkType == hyperv1.OVNKubernetes &&
+		guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig != nil &&
+		guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig.MTU != nil &&
+		*guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig.MTU < recommendedMinMTU-ovnkOverhead {
```

Comment:
For consistency, I think we should be setting this condition to indicate that a valid MTU is detected as well. Meaning, not just setting the condition when a suboptimal condition is found.

This aligns our usage of conditions with how the rest of the HCP conditions appear to work. Typically we have a condition that checks if something is `valid` and returns a status of `true` if the thing is valid, or `false` with a unique reason if the thing is not valid.